test/framework/robot.py

```diff
@@ -567,6 +567,14 @@ def test_get_toolchain_hierarchy(self):
             {'name': 'gompi', 'version': '1.4.10'},
```

Comment:
I don't think `gompi` is a broken choice, because we always include the *toolchain* too, not only the deps.

So, `dummy` is considered as a subtoolchain because `GCC` is built with `dummy`